### PR TITLE
Add libvirt support

### PR DESCRIPTION
**Pull request by fsimonis**
This PR adds a libvirt box to the preCICE VM.

The workflow is currently untested.

### CODE REVIEW

**Comment by fsimonis** [comment]
I was able to build this box locally after applying #28.
Without the patch, I ran repeatedly into download issues with the OpenFOAM packages

**Comment by fsimonis** [comment]
New issue. The libvirt provider is not available on windows.
So we need self-hosted runners here.

**Comment by fsimonis** [comment]
Looks like enabling libvirt on macOS is not as easy as I initially thought.
This workflow would be a good candidate for a self-hosted runner.

Anyhow, there is functionally no difference between the virtualbox and the libvirt-box.
We could switch the image and then build the libvirt-box manually until we have a self-hosted runner.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -21,11 +28,29 @@ jobs:
...
+
+
     - name: Show Vagrant version
       run: vagrant --version
 
     - name: Run vagrant up
-      run: vagrant up
+      run: |
+        virsh version
+        libvirtd --help
```
Comment (MakisH) [inline .github/workflows/build.yml:52]: Is this here only for debugging purposes? Same above.

File: Vagrantfile [hunk 2]
```diff
@@ -2,8 +2,9 @@
 # vi: set ft=ruby :
 
 Vagrant.configure("2") do |config|
-  # The Vagrant documentation recommends the bento images instead of ubuntu (more providers, open source)
-  config.vm.box = "bento/ubuntu-20.04"
+  # The generic/ images support virtualbox as well as libvirt and hyperv.
+  # This allows us to create performance oriented images for Linux (libvirt) and Windows (hyperv).
+  config.vm.box = "generic/ubuntu2004"
```
Comment (MakisH) [inline Vagrantfile:7]: Remember to also update `README.md` on this ("What is included?" section)

**Comment by MakisH** [comment]
I would first like some more detailed comparison between `bento/ubuntu-20.04` and `generic/ubuntu2004`, as this a fundamental change that can have side-effects.

Other than that, I am fine merging the rest.

**Comment by fsimonis** [comment]
I pulled the latest boxes and created diffable package lists of [bento](https://github.com/precice/vm/files/8334577/bento.txt) and [generic](https://github.com/precice/vm/files/8334578/generic.txt).

The sources for the [bento](https://github.com/chef/bento/blob/main/packer_templates/ubuntu/ubuntu-20.04-amd64.json) and the [generic](https://github.com/lavabit/robox/tree/master/scripts/ubuntu2004) box are open.


The generic image is slightly smaller, but my guess is that this doesn't really matter once we install all the basics.

**Review by MakisH — APPROVED** [review]
Aright, then let's merge it. I don't see any package that we need and would not anyway be installed, I don't know if there are any differences in configuration that we should consider, but let's find out.

Do you also need a release on Vagrant Cloud for any reason?

Please check my review comments above before merging.

File: .github/workflows/build.yml (hunk 1 above)
Comment (fsimonis) [reply .github/workflows/build.yml:52]: This is all debugging. I need to Cleanup this entire thing.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -21,11 +28,29 @@ jobs:
...
           ${{ runner.os }}-vagrant-
 
+    - name: Install libvirt
+      if: matrix.provider == 'libvirt'
+      run: |
+        brew install qemu libvirt
+        # disable unsupported security features
+        echo 'security_driver = "none"' >> /usr/local/etc/libvirt/qemu.conf
+        echo "dynamic_ownership = 0" >> /usr/local/etc/libvirt/qemu.conf
+        echo "remember_owner = 0" >> /usr/local/etc/libvirt/qemu.conf
```
Comment (MakisH) [inline .github/workflows/build.yml:38]: @fsimonis This needs some additional context in the comments

Comment (fsimonis) [reply .github/workflows/build.yml]: This was macOS specific and is now gone

**Comment by fsimonis** [comment]
The bento box for ubuntu 24.04 now supports libvirt and virtualbox.

**Pull request merged**